thorlcr/activities/enth/thenthslave.cpp

```diff
@@ -69,36 +68,35 @@ class BaseEnthActivity : public CSlaveActivity, implements ILookAheadStopNotify
     }
     void setLocalCountReq()
     {
+        ThorDataLinkMetaInfo info;
+        input->getMetaInfo(info);
         // Need lookahead _unless_ row count pre-known.
         if (0 == numerator)
             localRecCount = 0;
-        else if (intoMetaInfo.totalRowsMin == intoMetaInfo.totalRowsMax)
+        else if (info.totalRowsMin == info.totalRowsMax)
         {
-            localRecCount = (rowcount_t)intoMetaInfo.totalRowsMax;
+            localRecCount = (rowcount_t)info.totalRowsMax;
             ActPrintLog("%s: row count pre-known to be %" RCPF "d", actStr.str(), localRecCount);
         }
         else
+        {
             localRecCount = RCUNBOUND;
+            IStartableEngineRowStream *lookAhead = createRowStreamLookAhead(this, inputStream, queryRowInterfaces(input), ENTH_SMART_BUFFER_SIZE, true, false, RCUNBOUND, this, &container.queryJob().queryIDiskUsage());
```

Comment:
If this was used inside a global loop then lookahead would be repeatedly set, and may have problems because inputStream will be the previous look ahead.
Same goes for the global firstn below.